saleor/tests/e2e/checkout/test_checkout_calculate_correct_discount_for_sale_and_voucher.py

```diff
@@ -0,0 +1,201 @@
+import pytest
+
+from ..product.utils.preparing_product import prepare_product
+from ..sales.utils import create_sale, create_sale_channel_listing, sale_catalogues_add
+from ..shop.utils.preparing_shop import prepare_shop
+from ..utils import assign_permissions
+from ..vouchers.utils import create_voucher, create_voucher_channel_listing
+from .utils import (
+    checkout_add_promo_code,
+    checkout_complete,
+    checkout_create,
+    checkout_delivery_method_update,
+    checkout_dummy_payment_create,
+    checkout_lines_add,
+)
+
+
+def prepare_sale_for_variant(
+    e2e_staff_api_client,
+    channel_id,
+    product_variant_id,
+    sale_discount_type,
+    sale_discount_value,
+):
+    sale_name = "Sale"
+    sale = create_sale(e2e_staff_api_client, sale_name, sale_discount_type)
+    sale_id = sale["id"]
+    sale_listing_input = [
+        {
+            "channelId": channel_id,
+            "discountValue": sale_discount_value,
+        }
+    ]
+    create_sale_channel_listing(
+        e2e_staff_api_client, sale_id, add_channels=sale_listing_input
+    )
+    catalogue_input = {"variants": [product_variant_id]}
+    sale_catalogues_add(e2e_staff_api_client, sale_id, catalogue_input)
+
+    return sale_id, sale_discount_value
+
+
+def prepare_voucher(
+    e2e_staff_api_client,
+    channel_id,
+    voucher_code,
+    voucher_discount_type,
+    voucher_discount_value,
+):
+    voucher_data = create_voucher(
+        e2e_staff_api_client,
+        voucher_discount_type,
+        voucher_code,
+        "ENTIRE_ORDER",
+    )
+    voucher_id = voucher_data["id"]
+    channel_listing = [
+        {"channelId": channel_id, "discountValue": voucher_discount_value}
+    ]
+    create_voucher_channel_listing(e2e_staff_api_client, voucher_id, channel_listing)
+
+    return voucher_discount_value, voucher_code
+
+
+@pytest.mark.e2e
+@pytest.mark.parametrize(
+    "variant_price, sale_discount_value, sale_discount_type, expected_sale_discount, "
+    "voucher_discount_value, voucher_discount_type, expected_voucher_discount, "
+    "expected_unit_price",
+    [
+        ("19.99", 13, "PERCENTAGE", 2.60, 13, "PERCENTAGE", 2.26, 15.13),
+        ("30", 10, "FIXED", 10, 5, "FIXED", 5, 17.5),
+        ("56.50", 19.99, "FIXED", 19.99, 33, "PERCENTAGE", 12.05, 24.46),
+        ("77.77", 17.5, "PERCENTAGE", 13.61, 25, "FIXED", 25, 51.66),
+    ],
+)
+def test_checkout_calculate_discount_for_sale_and_voucher_1014(
+    e2e_not_logged_api_client,
+    e2e_staff_api_client,
+    permission_manage_products,
+    permission_manage_channels,
+    permission_manage_shipping,
+    permission_manage_product_types_and_attributes,
+    permission_manage_discounts,
+    variant_price,
+    sale_discount_value,
+    sale_discount_type,
+    expected_sale_discount,
+    voucher_discount_value,
+    voucher_discount_type,
+    expected_voucher_discount,
+    expected_unit_price,
+):
+    # Before
+    channel_slug = "test-channel"
```

Comment:
You don't have to set up channel_slug here since it is returned by prepare_shop() in line 106